### PR TITLE
DDB 1150 - Delete measure

### PR DESCRIPTION
**Pull request by leighjohnson-friday**
Allow an admin user to delete a measure 
also deletes RAYG row when measure is not in a group

### CODE REVIEW

File: pages/data-entry-entity/measure-delete/MeasureDelete.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      flash
+    ];
+  }
+
+  async postRequest(req, res) {
+    try {
+      await this.deleteMeasure();
+    } catch (error) {
+      logger.error(error);
+      req.flash(["An error occoured when deleting the measure."]);
```
Comment (dtaylor7) [inline pages/data-entry-entity/measure-delete/MeasureDelete.js:47]: as far as i can tell this wont get printed on page, you will need something like in template:
```
{% if flash %}
      <div class="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" tabindex="-1" data-module="govuk-error-summary">
        <h2 class="govuk-error-summary__title" id="error-summary-title">Error</h2>
        <div class="govuk-error-summary__body">
          <ul class="govuk-list govuk-error-summary__list">
            <li>
              <a href="#headline-measures">{{ flash }}</a>
            </li>
          </ul>
        </div>
      </div>
    {% endif %}
```

File: pages/data-entry-entity/measure-delete/MeasureDelete.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      // if measure is the only item in the group, remove the RAYG row
+      if (uniqMetricIds.length === 1) {
+        for (const entity of raygEntities) {
+          await Entity.delete(entity.id, { transaction });
+        }
+      }
+
+      await transaction.commit();
+    } catch (error) {
+      logger.error(error);
```
Comment (dtaylor7) [inline pages/data-entry-entity/measure-delete/MeasureDelete.js:73]: this error is also logged here https://github.com/alphagov/co-eu-transition-dashboard/pull/397/files#diff-0458da413e6e25b246b40872f59923a0296737c2e70595c01273c2ff604d6404R46

File: pages/data-entry-entity/measure-delete/MeasureDelete.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      raygEntities,
+      uniqMetricIds
+    };
+  }
+
+  async getMeasureData() {
+    const { measureEntities } = await this.getMeasure();
+
+    // measuresEntities are already sorted by date, so the last entry is the newest
+    const latestEntity = measureEntities[measureEntities.length - 1];
```
Comment (dtaylor7) [inline pages/data-entry-entity/measure-delete/MeasureDelete.js:147]: fyi: could also use `measureEntities.pop()`

File: pages/data-entry-entity/measure-edit/partials/forms.njk [hunk 4]
```diff
@@ -48,3 +48,11 @@
 {{ govukTabs({
   items: items
 }) }}
+
+{% if isUserAdmin and not preventDeleteForGroupMeasure %}
```
Comment (dtaylor7) [inline pages/data-entry-entity/measure-edit/partials/forms.njk:52]: you can access `req` inside template, so can be written as `req.user.isAdmin`

**Review by dtaylor7 — CHANGES_REQUESTED** [review]
nice work again, couple of FYI and optonals, think only required on is printing flash message to page if error

File: pages/data-entry-entity/measure-delete/MeasureDelete.js (hunk 1 above)
Comment (leighjohnson-friday) [reply pages/data-entry-entity/measure-delete/MeasureDelete.js]: The error var was getting set at the top of the page but updated to make it cleaner.

File: pages/data-entry-entity/measure-delete/MeasureDelete.js (hunk 3 above)
Comment (leighjohnson-friday) [reply pages/data-entry-entity/measure-delete/MeasureDelete.js]: Done

File: pages/data-entry-entity/measure-edit/partials/forms.njk (hunk 4 above)
Comment (leighjohnson-friday) [reply pages/data-entry-entity/measure-edit/partials/forms.njk]: Done

**Review by dtaylor7 — APPROVED** [review]

**Pull request merged**